djangocms_bootstrap4/contrib/bootstrap4_grid/cms_plugins.py

```diff
@@ -0,0 +1,187 @@
+# -*- coding: utf-8 -*-
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+
+from cms.models import CMSPlugin
+from cms.plugin_base import CMSPluginBase
+from cms.plugin_pool import plugin_pool
+
+from ...helpers import concat_classes
+from ...constants import DEVICE_SIZES
+
+from .models import (
+    Bootstrap4GridContainer,
+    Bootstrap4GridRow,
+    Bootstrap4GridColumn,
+)
+from .forms import Bootstrap4GridRowForm, Bootstrap4GridColumnForm
+
+
+class Bootstrap4GridContainerPlugin(CMSPluginBase):
+    """
+    Layout > Grid: "Container" Plugin
+    https://getbootstrap.com/docs/4.0/layout/grid/
+    """
+    model = Bootstrap4GridContainer
+    name = _('Container')
+    module = _('Bootstrap 4')
+    render_template = 'djangocms_bootstrap4/grid_container.html'
+    allow_children = True
+
+    fieldsets = [
+        (None, {
+            'fields': (
+                'container_type',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'tag_type',
+                'attributes',
+            )
+        }),
+    ]
+
+    def render(self, context, instance, placeholder):
+        classes = concat_classes([
+            instance.container_type,
+        ], instance.attributes)
+        instance.attributes['class'] = classes
```

Comment:
I suggest to explicitly declare the classes in the template and pass them via context.
Also change the signature of concat_classes to not take attributes:

```python
def concat_classes(classes, last_class=None):
    ...

context['classes'] = concat_classes(
    [instance.container_type],
    last_class=instance.attributes.get('class),
)
```